### ISSUE TITLE
[UX] Reorganize the admin bar to be more intuitive

### ISSUE THREAD

**Issue by sutibun** [backdrop/backdrop-issues#1149]
Maybe it's because I haven't used the admin bar for long but I don't find its organization intuitive at all.
## Current Admin bar

![backdrop-top-menu](https://cloud.githubusercontent.com/assets/14132746/9841674/6bed251e-5a68-11e5-8568-57b24e28daf7.png)

Current organization is as follows
- **Content**
  - Add content
  - Find content
  - Comments
- **User Accounts**
  - Add user account
  - Find user accounts
- **Appearance**
  - Install theme
  - Themes
  - Update
- **Functionality**
  - Install module
  - Modules
  - Update
  - Uninstall
- **Structure**
  - Content types
  - Custom blocks
  - Layouts
  - Menus
  - Taxonomy
  - Views
- **Config**
  - Administration
  - Content Authoring
  - Development
  - Media
  - Regional and Language
  - Search and metadata
  - System
  - User Accounts
  - Web services
- **Reports**
## Reasoning behind reorganizing the admin bar

Here is what I had in mind (attached with my thinking of the changes)
- **Content**: If you want to manage any aspect of the site's content, this feels like THE place to go to.
  - Add content
  - Find content
  - Comments
  - _Content types_: The wording itself calls for this: "Content > _Content_ types". When I am under the "Content" section, I _should be able to_ look over comments (a type of content), create new pieces of content, search for past content pieces, AND edit the types of content I can create. Content. Content. Content. Anything dealing with the management of content feels like it should be grouped together and be near each other. Feels more intuitive.
  - _Taxonomy_: Taxonomy is all about classifying/organizing/tagging ... say it with me... _content!_ If that's not the management of content, I don't know what is. It doesn't make sense to manage a content's category/tags elsewhere when you have a section specifically for content management. Even Wordpress' taxonomy is under post types. Looking under the WP Post type on the side nav, you have the following under it: "All Posts", "Add New", "Categories", "Tags".
  - _Custom blocks_: Honestly, this I am not sure where this belongs whether here or in the Appearance section. But seems like blocks are for content lacking URLS... I am leaning towards content.
- **User Accounts**
- **Appearance**: Anything dealing with the look of the site goes here, at least that's the way I see it.
  - Install theme
  - Themes
  - _Layouts_: You can't create a theme without Layouts in Backdrop. Layouts and themes are meant to work together so they should be close together. Plus, Layouts actually change the _appearance_ of the site which is more reason for it to be under Appearance.
  - _Menus_: What's under Menus? Account menu, Admin menu, Primary nav. They all influence the appearance of the site. Account + Admin influence the backend appearance while the primary nav influences the frontend look. Appearance. Appearance. Appearance. (sorry)
  - _Views_: This manipulates the appearance of content listings (HTML list, grid, etc) I even remember reading earlier a suggestion to rename Views to Display. The name themselves deal with Appearances.
  - Update: This is me being nitpicky... but do people actually update themes through this? It seems like a duplicate of "Functionality>Update", which I suspect most use to update anything in their CMS, including themes. Feels like there should be one central place for updating. Suggest removing this but maybe I am missing something?
- **Functionality**
  - Install module
  - _Modules_: Why do we have duplicates of "Modules" and "Functionality"? They lead to the same thing except they have different URLs. Also, since we're on this topic, would it better to rename modules to plugins? Two big softwares use the term "plugin": Wordpress and Firefox. Just wondering...
  - Update 
  - Uninstall
- **Config**
- **Reports**

In other words, goodbye "Structure". Or, that can be kept for contrib modules or something else I am overlooking...
## Simpler and improved admin bar?

For the sake of clarity, here is my suggested admin bar without my comments
- **Content**
  - Add content
  - Find content
  - Comments
  - _Content types_
  - _Taxonomy_
  - _Custom blocks_
- **User Accounts**
- **Appearance**
  - Install theme
  - Themes
  - _Layouts_
  - _Menus_
  - _Views_
- **Functionality**
  - Install module
  - Modules
  - Update 
  - Uninstall
- **Config**
- **Reports**

And just because I like saying it. Content. Content. Content. ^_^)

**Comment by Graham-72** [comment]
I agree that we need improvement to the organization of the admin bar. I was just thinking this the other day when I got tired of going to one place to make a backup and a different place to put my site into maintenance mode when doing updates.

**Comment by docwilmot** [comment]
I like.

But I think Layouts, Views, Custom Blocks, and Menus are more about the organization of your site rather than being pure content or appearance. I'd keep "Structure" for those, even if with a better label.

And the "Configuration" group is a mess I agree. There was an issue for that somewhere.

**Comment by quicksketch** [comment]
I think Content Types used to be under Content in something like the Drupal 6 era. They were moved because "Content" is generally one role of responsibility (editors, content creators, bloggers, etc) but "Content Types" is a site-builder responsibility and something that is done when the site is created (along with Views, Layouts, etc). As for whether it's more or less intuitive, it's hard to say.

I also think lumping Views and Layouts into Appearance might be a little aggressive. I get the feeling that the elimination of "Structure" was the primary goal here, but the end arrangement makes some matches that I don't feel are entirely better.

I do agree about Custom Blocks and Terms being "content". Especially now that blocks are separated from layout, custom blocks really are just content without a URL. Terms are frequently an editorial task as well, in the event that a new category needs to be added.

> Also, since we're on this topic, would it better to rename modules to plugins?

Although it might be nice if we'd always called modules "plugins" from the start, that switch would be much too painful for very little gain. Apache uses "modules", PHP uses "extensions", WordPress "plugins", Joomla has *both* "plugins" *and* "modules", and of course Backdrop/Drupal use "modules". (Firefox also btw uses "Extensions" not "Plugins"). I don't think any term is substantially better than another here, and the impact of such of change wouldn't be justifiable, IMO.

For a recap of a bunch of other renamings we did (including Functionality), see the meta issue at https://github.com/backdrop/backdrop-issues/issues/127.

To make this more actionable, I think it would be great if we could split this into separate issues and have a meta to group them all together.

**Comment by sutibun** [comment]
First, a warning. It's late here and I am finding it slightly difficult to get across my meaning in words. Hope the following makes some sense.

Damage control :p
-----

@quicksketch
> I get the feeling that the elimination of "Structure" was the primary goal here...

Not at all. I think my comment

> In other words, goodbye "Structure". Or, that can be kept for contrib modules or something else I am overlooking...

might've given off that vibe but that wasn't my intention.

When I finished arranging the menu items how I saw fit, there was nothing left for "Structure". That's when I added that comment there before moving onto the next section. Probably should've left it out but I had just spent some time writing this long piece and was kinda bored so adding these comments is my way to make it fun. Makes writing a lil' better when adding my personality.

Bold or Silly? Extrapolating from a small sample to the masses
-----

Anyway, let me explain further why I moved Layout, Views, and Menus under Appearance. I think part of it stems from the way I work with Drupal (And I guess now Backdrop)

With a new site, I go right into building it. There's some planning but most of it in my head and it doesn't take long. I take the "get-it-out-there-already" approach.

Builders who plan for months and have figured out everything they'll need from the start are either professional Drupal guys or companies with budget. They think logically... structure, taxonomy, etc. Basically, they get Drupal. It makes sense to have a role for "site builder", "content manager", etc.

Not sure that's how others see it.

A month ago, I was helping a tiny non-profit with their WP site. The original volunteer who had built it was gone. Everyone had their hands on the WP admin. There were no designated roles.

The interesting thing is all their questions were phrased in terms of outside-looking in. A kind of WYSIWYG approach.

- can you remove that line?
- can you add this file over there?
- Will updating the menu on top be easy?

No one is thinking in terms of structure, functionality, etc. 

On top of that, they were building as we go. We only met once and will meet in a few months? How many of them will remember how WP works still?

Less sophisticated guys, like me, don't have the time nor budget for extensive planning. So, we're always going in and out of development with big time gaps in between.

I'm not too different from the "WYSIWYG" approach.

As an example, let's say, I am starting with a brand new site. I create an Event content type and a View to list them plainly. I add other stuff, finish building out the initial site and hit publish.

Sweet.

Time passes, and I realize I want to add a new section to the site that **displays** the event info in a calendar-style. Because of the time gap, I am NOT in the Drupal mentality. No, when I return, my mindset is on the end result. The WYSIWYG approach. I want calendar boxes. How do I get that? Structures? WTF. I just want to change how this looks.

To do the calendar, I'd imagine I would go to Views to massage the fields, possibly even use View grids, use CSS injector to add my custom styling (or go to themes, assuming Backdrop is much easier to design in). And then, in the end, I'd remember to change the primary nav to link to the new page. (I tend to forget about the menus last)

Wheww. Done.

Months pass and now find I want to change how my blog **looks** like. Once again, I am not in the Backdrop mentality thinking in terms of structure. All I'm thinking about is what I see on the site... its appearance and what I envision. In this case, what I want is to drop the standard blog look (two column layout) and switch to a magazine-style design with excerpts.

I imagine I would go to Views and change some fields. Next, go to Layouts and then update the CSS file (theme).

**Whenever I want to visually tweak something on my sites, I find myself using Themes/CSS Injector, Views, Panels/Layout, Menu, etc.**

This is how I work. That's why I associate them with Appearances.

Breakdown
--------

I get it.

I see why logically and conceptually those would best fit under a section called "Structure". Just like it makes sense for biologists to group organisms and give them these weird names. But do normal people think that way.

**Really, what does the word structure mean to a normal user?**

I'd say they are more likely to think and behave like I do. They view things from the front end... like a WYSIWYG editor. They don't think in terms of structure.

> I want my about page to have one column. How do I do this?

Go to Appearance > Layouts.

> How do I make it so I can add a new link to the top page?

Go to Appearance > Menu

> I want my blog to list recent posts. The first post should display only the featured image but the rest simply excerpts. Can Backdrop do this?

Go to Appearance > Views.

> I want a responsive design. Where do I go?

Go to Appearance > Themes.

> I want to change how my site looks like but it's complex. Where do I go? Appearance?

You got it. Everything you need to alter the look is under the Appearance section. If not, you'll need to check out contrib modules.

I like what cellear said https://github.com/backdrop/backdrop-issues/issues/209#issuecomment-49703623

> Appearance is more than just themes.

When I need to customize my site's look in some shape or form, Appearance feels like the one-stop place to go to. The starting point.

Compressing it all down to a few sentences
-----

If I can say all that in 4 lines

1. Whenever you use Layouts, Views, and Menus, you are essentially changing the appearance of the site. *Their names themselves imply design-related stuff.**
2. Normal users, and myself when going in and out, think in terms of the look of a site. Not in Drupal/Backdrop speak. The perspective is from the frontend.
3. To change how the front end appears, you tend to use Themes, Layouts, Views, and Menus.
4. So, from that point of view, it's best to keep them together under Appearance.

**Comment by sutibun** [comment]
Simple thinking exposed
-----
The same thing with content type.

When I return to Drupal, there's a time gap from the last time I used it to build out the site. So, my initial mindset going is in "reset" and "fresh" mode some times. The experience is along the lines of

> I need to create a new type of *content*. Well, it's *content-related* so it should be under the content menu. … Oh wait, in Drupal it's under … structure.

Structure… to me has no meaning.

I kind of expect it to be under content but then get jolted when I remember Drupal has it's own categorization going on. I find myself at times using the search form in the admin bar. Much easier than remembering where things are.

I guess for me, I think of just 3 overall universes/categories

- Content - anything dealing within the realm of creating or editing content should be here.
- Design - if I want to change how I present that content, how to add a new site section, or update the style, I think of appearance.
- Admin/Settings - Anything dealing with nit-picky details, goes here.

I don't know if this way of organizing is the best but I find it more comfortable and requires less thinking involved than the current setup.

Disrupting your own preferences?
----

On the other hand, Wordpress *seems* more content creation-focused. If looking at the site navigation, they have content links kind of above the fold line followed by appearance and admin stuff on the bottom

- post types
- dashboard
- comment area
- media
- pages
- appearance 
- comments
- plugins
- settings

From a birds eye view:

- content-related links lumped above
- one link to design stuff
- admin and other plugin stuff below

There's no structure, no user, etc. all tucked away. Food for thought.

Maybe as an alternate idea, **what if we don't need to emphasize everything on the admin bar?** What if we also develop a better dashboard that a user lands on every time they login and has every day stuff on it.

**Comment by klonos** [comment]
> ...As for whether it's more or less intuitive, it's hard to say.

> I also think lumping Views and Layouts into Appearance might be a little aggressive. I get the feeling that the elimination of "Structure" was the primary goal here, but the end arrangement makes some matches that I don't feel are entirely better.

I'm as skeptical about this as Nate is.

> They view things from the front end... like a WYSIWYG editor. They don't think in terms of structure.

But I fully understand where you're coming from here. You have a point if you look at it from an end-user point of view.

> I take the "get-it-out-there-already" approach.
> ...
> Not sure that's how others see it.

I'm like you! :wink: 

> Screwing with your mind ...

No man. Please don't! :stuck_out_tongue_winking_eye:

**Comment by sutibun** [comment]
Somehow found myself on Joomla's website. Took a look over their demo site and noticed their menus. Thought it'd be interesting to post how they went about it as a different POV.

Joomla top menu
---

![joomla-menu](https://cloud.githubusercontent.com/assets/14132746/9841585/3691cac4-5a67-11e5-9c7b-e43498faa52b.png)

- **System**
  - Control Panel
  - Global Configuration
  - Global Check-in
  - Clear Cache
  - Clear Expired Cache
  - System Information
- **Users**
  - Manage
  - Groups
  - Access Levels
  - User Notes
  - User Notes Categories
  - Mass Mail Users
- **Menus**
  - Manage
  - Main Menu
  - User Menu
- **Content**
  - Articles
  - Categories
  - Featured Articles
  - Media
- **Components**
  - Banners
  - Contacts
  - Joomla Update
  - Messaging
  - News Feeds
  - Post-installation Messages
  - Redirect
  - Search
  - Smart Search
  - Tags
- **Extensions**
  - Manage
  - Modules
  - Plugins
  - Templates
  - Language(s)
- **Help**

Joomla side menu
----

They also have a side menu on what looks like the dashboard.

![joomla-side-menu](https://cloud.githubusercontent.com/assets/14132746/9841586/369ce436-5a67-11e5-84e4-2e75e37b6772.png)

**Comment by jenlampton** [comment]
> I think Content Types used to be under Content in something like the Drupal 6 era. They were moved because "Content" is generally one role of responsibility (editors, content creators, bloggers, etc) but "Content Types" is a site-builder responsibility and something that is done when the site is created (along with Views, Layouts, etc). As for whether it's more or less intuitive, it's hard to say.

Back in Drupal 6 this was made more clear because the 'Structure' menu was also called 'Site building'.

**Comment by MrHaroldA** [comment]
I actually had to search for 'admin/structure/taxonomy' today, expecting it to be at the Content main menu item.

**Comment by BWPanda** [comment]
Duplicate of #2118 (or the other way around)?

**Comment by BWPanda** [comment]
Closing this as a duplicate of https://github.com/backdrop/backdrop-issues/issues/2118 which seems to have more (recent) discussion.
